OsmAnd/src/net/osmand/plus/resources/ResourceManager.java

```diff
@@ -99,6 +100,7 @@
 	private List<TilesCache> tilesCacheList = new ArrayList<>();
 	private BitmapTilesCache bitmapTilesCache;
 	private GeometryTilesCache geometryTilesCache;
+	private Map<MapTileLayer, Integer> mapTileLayerSizes = new ConcurrentHashMap<>();
```

Comment:
List<MapTileLayerSize> but always recreate if new element added or deleted
MapTileLayerSize { MapTileLayer, int, markToGCTimestamp=null / date, activeTimestamp} 

- delete in update if it wasn't set for 60 sec (first marktogc / 2nd delete)